bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java

```diff
@@ -136,7 +136,7 @@ public Integer getRank() {
                     locale);
             if (dynamicStateDescription != null) {
                 // Compare by reference to make sure a new state description is returned
-                if (dynamicStateDescription == originalStateDescription) {
+                if (dynamicStateDescription.equals(originalStateDescription)) {
```

Comment:
Look at the comment, this is intentionally `==`.

Reply:
🙈 sorry